### PR TITLE
robust for url compuation of 'view raw'

### PR DESCRIPTION
**Pull request by richardo2016**
Thank you for creating a pull request to contribute to Hexo code! Before you open the request please review the following guidelines and tips to help it be more easily integrated:

- [x] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

File: lib/plugins/tag/include_code.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ module.exports = function(ctx) {
       // If the language is not defined, use file extension instead
       lang = lang || pathFn.extname(path).substring(1);
 
-      caption = '<span>' + title + '</span><a href="' + ctx.config.root + codeDir + path + '">view raw</a>';
+      caption = '<span>' + title + '</span><a href="' + pathFn.join(ctx.config.root, codeDir, path) + '">view raw</a>';
```
Comment (JLHwung) [inline lib/plugins/tag/include_code.js]: The `join` will generate `\\` on Windows platform, but `\\` is not a valid URI hierarchy separator, see [RFC 1738 Section 2.3](https://www.ietf.org/rfc/rfc1738.txt).

**Comment by tcrowe** [comment]
For URL join I usually use this: https://github.com/jfromaniello/url-join

I'm not recommending that we add it as a dependency but just recognize that to join URLs can be complex.

File: lib/plugins/tag/include_code.js (hunk 1 above)
Comment (stevenjoezhang) [reply lib/plugins/tag/include_code.js:67]:
```
pathFn.posix.join
```

**Comment by stevenjoezhang** [comment]
Superseded by https://github.com/hexojs/hexo/pull/4996

**Pull request closed without merging**